### ISSUE TITLE
feat(modernbert): port the ModernBERT encoder (alternating local/global attention, RoPE, GeGLU) for /v1/embeddings

### ISSUE THREAD

**Issue by inureyes** [lablup/mlxcel#1332]
## Summary

ModernBERT (`model_type: modernbert`) is an 8192-context encoder with pre-norm blocks, RoPE instead of absolute positions, a fused `Wqkv` projection, a GeGLU MLP, and an attention pattern that alternates two local (sliding-window, base 10000) layers with one global (base 160000) layer. mlxcel has none of this. This issue adds `src/models/modernbert.rs` with the embedder and the `ModernBertForSequenceClassification` head so nomic-embed / gte-modernbert checkpoints serve through `/v1/embeddings` and gte-reranker-modernbert serves through #1356.

## Current behavior

- `src/models/detection.rs:124-199` has no `"modernbert"` arm.
- The only bidirectional sliding-window mask code in the tree is causal: `create_causal_mask_with_window` (`src/lib/mlxcel-core/src/utils.rs:558`). #1353 adds `create_bidirectional_window_mask`, which this port consumes.
- `mlxcel_core::fast_rope` (`src/lib/mlxcel-core/src/lib.rs`) already implements the non-traditional RoPE ModernBERT uses (full `head_dim`, `traditional = false`, configurable base).

### Config (`config.json`, verified on `nomic-ai/modernbert-embed-base` and `Alibaba-NLP/gte-reranker-modernbert-base`)

| key | value / default |
|---|---|
| `vocab_size` | 50368 |
| `hidden_size` | 768 (`D`) |
| `num_hidden_layers` | 22 |
| `num_attention_heads` | 12 (`head_dim = 64`) |
| `intermediate_size` | 1152 |
| `max_position_embeddings` | 8192 |
| `norm_eps` | 1e-5 (`layer_norm_eps` is an alias) |
| `norm_bias` | false |
| `global_rope_theta` | 160000.0 |
| `local_rope_theta` | 10000.0 (null means use global) |
| `global_attn_every_n_layers` | 3 |
| `local_attention` | 128 (window, each side `local_attention / 2`) |
| `attention_bias`, `mlp_bias` | false |
| `hidden_activation` | `gelu` (exact, `mlxcel_core::gelu`) |
| `pad_token_id` | 50283 |
| `classifier_pooling` | `cls` or `mean` (classifier only) |
| `classifier_activation` | `gelu` (only value supported) |
| `classifier_bias` | false |
| `num_labels` / `id2label` | head width |

Detection: `"modernbert" => ModelType::ModernBert`; `architectures[0] == "ModernBertForSequenceClassification"` selects the classifier; `ModernBertModel` and `ModernBertForMaskedLM` are embedders.

### Forward

```
h = LayerNorm(tok_embeddings[input_ids])                      # embeddings.norm, eps norm_eps, no bias
global_mask  = create_bidirectional_padding_mask(attention_mask)            # [B,1,1,L]
sliding_mask = create_bidirectional_window_mask(attention_mask, local_attention/2 + 1)
               # blocked iff padding or |q - k| > local_attention/2   (|q-k| <= 64 attends)
for i in 0..num_hidden_layers:
    local = (i % global_attn_every_n_layers) != 0
    x = h if i == 0 else LayerNorm(h)                          # layers.0 has no attn_norm
    qkv = Wqkv(x)                                              # [B, L, 3D]
    q, k, v = qkv[..., 0:D], qkv[..., D:2D], qkv[..., 2D:3D]   # each -> [B, heads, L, 64]
    base = local_rope_theta if local else global_rope_theta
    q = fast_rope(q, 64, traditional=false, base, 1.0, offset=0); k likewise
    attn = sdpa(q, k, v, scale = 64^-0.5, mask = sliding_mask if local else global_mask)
    h = h + Wo(attn)
    y = Wi(LayerNorm(h))                                       # mlp_norm; Wi: [2*inter, D]
    inp, gate = y[..., :inter], y[..., inter:]
    h = h + Wo_mlp(gelu(inp) * gate)
last_hidden_state = LayerNorm(h)                               # final_norm
embedding = normalize_l2(pool(last_hidden_state, attention_mask, mode))   # default mean
```

Classifier head:
```
pooled = h[:, 0] if classifier_pooling == "cls" else mean_pool(h, attention_mask)
logits = classifier(LayerNorm(gelu(head.dense(pooled))))      # classifier: [num_labels, D] with bias
```

### Weight keys and sanitize

```
[model.]embeddings.tok_embeddings.weight   [50368, 768]
[model.]embeddings.norm.weight             [768]
[model.]layers.{i}.attn.Wqkv.weight        [2304, 768]
[model.]layers.{i}.attn.Wo.weight          [768, 768]
[model.]layers.{i}.attn_norm.weight        [768]      (absent for i == 0)
[model.]layers.{i}.mlp.Wi.weight           [2304, 768]
[model.]layers.{i}.mlp.Wo.weight           [768, 1152]
[model.]layers.{i}.mlp_norm.weight         [768]
[model.]final_norm.weight                  [768]
head.dense.weight [768, 768], head.norm.weight [768], classifier.{weight,bias}   (classifier only)
decoder.{weight,bias}, head.* (MLM checkpoints: drop head.* for the embedder, drop decoder.* always)
```

Sanitize: strip a leading `model.`; drop `decoder.*` and `pooler.*`; for the embedder also drop `head.*` and `classifier.*`. `Wqkv` and `Wi` are loaded through `UnifiedLinear` (quantized checkpoints quantize them as single tensors; split after the projection, never split the weight).

### Limits and defaults

- Default pooling `mean`; `1_Pooling/config.json` overrides (nomic: mean).
- `max_length = min(sentence_bert_config max_seq_length, model_max_length, 8192)`; RoPE has no position table, so `max_position_embeddings` (8192) is the cap rather than a hard error.
- Tokenizer adds `[CLS]`/`[SEP]` through its post-processor; no `token_type_ids`.
- f32 checkpoints run in f32; bf16 follows the text rule.

## Implementation plan

1. `src/models/modernbert.rs` (+ `modernbert_tests.rs`): `ModernBertArgs` (serde, defaults above, `#[serde(alias = "layer_norm_eps")]` on `norm_eps`), `ModernBertLayer { attn_norm: Option<LayerNorm>, wqkv, wo, wi, wo_mlp: UnifiedLinear, mlp_norm: LayerNorm, local: bool, rope_base: f32 }`, `ModernBertEncoder { tok_embeddings: UnifiedEmbedding, embed_norm, layers, final_norm, num_heads, head_dim, local_window }` with `encode(input_ids, attention_mask) -> [B, L, D]`; `ModernBertEmbeddingModel` implementing `EmbeddingModel` (`default_pooling = Mean`, `embedding_dim = hidden_size`); `ModernBertSequenceClassifier { encoder, head_dense, head_norm, classifier, pooling }` with `logits(input_ids, attention_mask) -> [B, num_labels]`.
2. Use `mlxcel_core::layers::attention` with the explicit mask; `slice_axis` for the `Wqkv` and `Wi` splits; `mlxcel_core::gelu`; `create_bidirectional_window_mask` from #1353.
3. `src/models/detection.rs` arm, `src/model_metadata.rs` entry (`kind: Embedding`), `src/embeddings/loader.rs` arm.
4. `docs/supported-models.md` row; `docs/embeddings.md` note on nomic's `search_query: ` / `search_document: ` prefixes.

## Validation

(a) Unit tests in `src/models/modernbert_tests.rs`:
- `layer_parity_selects_local_and_global_rope_base` (with `global_attn_every_n_layers = 3`: layers 0, 3, 6 global; others local).
- `sliding_mask_attends_within_64_and_blocks_beyond` (L = 200, one padded row).
- `first_layer_has_no_attn_norm` (loader accepts a weight map without `layers.0.attn_norm.weight` and rejects one missing `layers.1.attn_norm.weight`).
- `geglu_splits_input_then_gate` (hand-built `Wi` where the gate half is zero gives a zero output).
- `padding_invariance` as in #1321.
- `classifier_mean_pooling_ignores_padding`.

(b) Real checkpoints:
```bash
./target/release/mlxcel download nomic-ai/modernbert-embed-base
./target/release/mlxcel embed -m models/modernbert-embed-base \
  -p "search_query: What is TSNE?" \
  -p "search_document: t-SNE is a dimensionality reduction technique used for visualizing high-dimensional data." \
  -p "search_document: The Eiffel Tower is a wrought-iron lattice tower in Paris."
./target/release/mlxcel-server -m models/modernbert-embed-base --port 8080
curl -s localhost:8080/v1/embeddings -H 'Content-Type: application/json' \
  -d '{"input": ["search_query: What is TSNE?", "search_document: t-SNE is a dimensionality reduction technique used for visualizing high-dimensional data."]}'
```
Acceptance: unit norms within 1e-5; cosine(query, t-SNE document) > cosine(query, Eiffel document) by at least 0.15; identical texts give cosine 1.0 within 1e-6; a 4096-token document embeds without error and its batched result matches the single-input result within 1e-3 (exercises local windows well past 128 tokens).

`Alibaba-NLP/gte-reranker-modernbert-base` is loaded through `ModernBertSequenceClassifier::load` and must produce finite `[B, 1]` logits; its ordering is validated in #1356.

## Acceptance criteria

- [x] `src/models/modernbert.rs` loads `ModernBertModel`, `ModernBertForMaskedLM` (MLM head dropped) and `ModernBertForSequenceClassification` from the key layout above. `ModernBertModel` and the classifier are verified on real checkpoints; the MLM path is covered by a synthetic sanitize test only, since no real `ModernBertForMaskedLM` checkpoint was available.
- [x] local layers use a bidirectional window of `local_attention / 2` on each side with `local_rope_theta`; global layers use the padding mask with `global_rope_theta`
- [x] nomic-ai/modernbert-embed-base passes the gates above through `mlxcel embed` and `POST /v1/embeddings`
- [x] docs/supported-models.md updated
- [x] detection table in src/models/detection.rs needed no change (#1353 already added the `modernbert` mapping and the encoder-only entry); the coverage is added as a test instead
- [x] test suite passes, run as `--profile test-fast --features cuda` on this Linux/CUDA host because `metal,accelerate` is macOS-only and unavailable here; macOS validation is still outstanding
- [x] `cargo clippy --profile test-fast --features cuda --lib --bins --tests -- -D warnings` and `cargo fmt --all -- --check` pass

## Out of scope

- Unpadded / flash-style packed attention (the HF fast path); mlxcel pads.
- `/v1/rerank` wiring (#1356).

## Dependencies

Blocked by #1353

- Part of epic #1348
